### ISSUE TITLE
Memory leakage

### ISSUE THREAD

**Issue by ALEXKIRNAS** [uber-research/jpeg2dct#4]
Hi! 
I have tried to use your lib to train network on DCT inputs and bumped into memory leakage issue.
I use Tensorflow operation to read DCT image from jpeg bytes. Here some useful info and dummy sample to reproduce the issue:

**System information**
- Have I written custom code: No
- OS Platform and Distribution: Linux Ubuntu 16.04.5 LTS (Xenial Xerus)
- TensorFlow installed from: source (v1.12.0-6120-gdaab267)
- TensorFlow version: 1.12.0
- Python version: 3.6.0
- CUDA/cuDNN version: CUDA 10.0.130 / CuDNN 7.0
- GPU model and memory: Tesla V100-SXM2 / 16Gb

**Packages information**
- numpy: 1.16.0
- libjpeg: libjpeg-turbo8-dev 1.4.2-0ubuntu3.1
- jpeg2dct installed from **pip** distribution (note, that issue reproducible on both pip and source installations) 

**Code to reproduce**
```python
from io import BytesIO

import PIL.Image as Image
import numpy as np
import tensorflow as tf
from jpeg2dct.tensorflow import decode


def img_to_bytes(image):
    image_pil = Image.fromarray(np.uint8(image))
    output = BytesIO()
    image_pil.save(output, format='JPEG')

    return output.getvalue()


def _rgb_to_dct_image(jpegbytes):
    dct_y, dct_cb, dct_cr = decode(jpegbytes)
    return dct_y, tf.concat([dct_cb, dct_cr], axis=-1)


def batch_rgb_to_dct_image(inputs):
    dct_y_batch, dct_cb_cr_batch = tf.map_fn(
        fn=_rgb_to_dct_image,
        elems=inputs,
        dtype=(tf.int16, tf.int16)
    )

    return dct_y_batch, dct_cb_cr_batch


images = tf.placeholder(shape=(32,), dtype=tf.string)
dct_y_batch, dct_cb_cr_batch = batch_rgb_to_dct_image(images)

real_images = np.random.uniform(size=(32, 1080, 1920, 3), low=0, high=255).astype(np.uint8)
images_strings = list(map(img_to_bytes, real_images))

sess = tf.Session()
while True:
    _, _ = sess.run([dct_y_batch, dct_cb_cr_batch], feed_dict={images: images_strings})

```

After several minutes script will be killed by OS due to extremely large memory usage.

**Comment by gueguenster** [comment]
what version of Pillow are you using? Pillow==5.0.0 has a memory leak issue (https://github.com/python-pillow/Pillow/issues/2972)

**Comment by gueguenster** [comment]
There is a memory leak indeed in the tensorflow section. Thanks for pointing that out. 
It does not seem to appear in the numpy section, as replacing the "while True" with:
```
from jpeg2dct.numpy import loads
while True:
    loads(images_strings[0])
```
seems to be allright

**Comment by gueguenster** [comment]
a new version has been released 0.2.4. I did not notice any memory leak after that modification. Let us know if you still face the same problem with that new version.

**Comment by ALEXKIRNAS** [comment]
Seems that the issue is solved! Thanks for quick response.
